pkg/apiserver/registry/vpc/vpc_test.go

```diff
@@ -0,0 +1,197 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package vpc_test
+
+import (
+	"fmt"
+
+	logger "github.com/go-logr/logr"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/internalversion"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+	"k8s.io/apiserver/pkg/endpoints/request"
+	"k8s.io/client-go/tools/cache"
+
+	runtimev1alpha1 "antrea.io/nephe/apis/runtime/v1alpha1"
+	. "antrea.io/nephe/pkg/apiserver/registry/vpc"
+	"antrea.io/nephe/pkg/controllers/cloud"
+)
+
+var _ = Describe("VPC", func() {
+	var vpcIndexer1 = cache.NewIndexer(
+		func(obj interface{}) (string, error) {
+			vpc := obj.(*runtimev1alpha1.Vpc)
+			return fmt.Sprintf("%v-%v", vpc.Annotations[cloud.VpcIndexerByAccountUUID], vpc.Info.Id), nil
+		},
+		cache.Indexers{
+			cloud.VpcIndexerByAccountNameSpacedName: func(obj interface{}) ([]string, error) {
+				vpc := obj.(*runtimev1alpha1.Vpc)
+				return []string{vpc.Annotations[cloud.VpcIndexerByAccountNameSpacedName]}, nil
+			},
+			cloud.VpcIndexerByVpcID: func(obj interface{}) ([]string, error) {
+				vpc := obj.(*runtimev1alpha1.Vpc)
+				return []string{vpc.Info.Id}, nil
+			},
+			cloud.VpcIndexerByNamespace: func(obj interface{}) ([]string, error) {
+				vpc := obj.(*runtimev1alpha1.Vpc)
+				return []string{vpc.Namespace}, nil
+			},
+			cloud.VpcIndexerByName: func(obj interface{}) ([]string, error) {
+				vpc := obj.(*runtimev1alpha1.Vpc)
+				return []string{vpc.Name}, nil
+			},
+		})
+
+	var l logger.Logger
+	cacheTest1 := &runtimev1alpha1.Vpc{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "default",
+			Name:      "targetId",
+		},
+		Info: runtimev1alpha1.VpcInfo{
+			Id:   "targetId",
+			Name: "targetName",
+		},
+	}
+	cacheTest2 := &runtimev1alpha1.Vpc{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "non-default",
+			Name:      "targetId-nondefault",
+		},
+		Info: runtimev1alpha1.VpcInfo{
+			Id:   "targetId-Non",
+			Name: "targetName",
+		},
+	}
+	cacheTest3 := &runtimev1alpha1.Vpc{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "default",
+			Name:      "targetId2",
+			Annotations: map[string]string{
+				cloud.VpcIndexerByAccountNameSpacedName: "default/accountname",
+			},
+		},
+		Info: runtimev1alpha1.VpcInfo{
+			Id:   "targetId2",
+			Name: "targetName",
+		},
+	}
+	cachedVpcs := []*runtimev1alpha1.Vpc{
+		cacheTest1,
+		cacheTest2,
+		cacheTest3,
+	}
+
+	expectedVpcs := []*runtimev1alpha1.Vpc{
+		cacheTest1,
+		nil,
+		cacheTest3,
+	}
+
+	Describe("Test Get function of Rest", func() {
+		It("Three status of realizing", func() {
+			for i, cachedVpc := range cachedVpcs {
```

Comment:
Did not get the test Name? What do you mean by "Three status of realizing"

Reply:
I think this is similar to virtualmachinepolicy_test, we have three tests to test three different situations, but maybe for now I can delete this for confusing